sympy/solvers/ode.py

```diff
@@ -1334,45 +1335,61 @@ class in it.  Note that a hint may do this anyway if
                 matching_hints["Liouville_Integral"] = r
 
         # Homogeneous second order differential equation of the form
-        # a3*f(x).diff(x, 2) + b3*f(x).diff(x) + c3, where
-        # for simplicity, a3, b3 and c3 are assumed to be polynomials.
+        # a3*f(x).diff(x, 2) + b3*f(x).diff(x) + c3
         # It has a definite power series solution at point x0 if, b3/a3 and c3/a3
         # are analytic at x0.
         deq = a3*(f(x).diff(x, 2)) + b3*df + c3*f(x)
         r = collect(reduced_eq,
             [f(x).diff(x, 2), f(x).diff(x), f(x)]).match(deq)
         ordinary = False
+        if r:
+            if not all([r[key].is_polynomial() for key in r]):
+                n, d = reduced_eq.as_numer_denom()
+                reduced_eq = expand(n)
+                r = collect(reduced_eq,
+                    [f(x).diff(x, 2), f(x).diff(x), f(x)]).match(deq)
         if r and r[a3] != 0:
-            if all([r[key].is_polynomial() for key in r]):
-                p = cancel(r[b3]/r[a3])  # Used below
-                q = cancel(r[c3]/r[a3])  # Used below
-                point = kwargs.get('x0', 0)
+            p = cancel(r[b3]/r[a3])  # Used below
+            q = cancel(r[c3]/r[a3])  # Used below
+            point = kwargs.get('x0', 0)
+            check = p.subs(x, point)
+            if not check.has(oo, NaN, zoo, -oo):
+                check = q.subs(x, point)
+                if not check.has(oo, NaN, zoo, -oo):
+                    ordinary = True
+                    r.update({'a3': a3, 'b3': b3, 'c3': c3, 'x0': point, 'terms': terms})
+                    matching_hints["2nd_power_series_ordinary"] = r
+
+            # Checking if the differential equation has a regular singular point
+            # at x0. It has a regular singular point at x0, if (b3/a3)*(x - x0)
+            # and (c3/a3)*((x - x0)**2) are analytic at x0.
+            if not ordinary:
```

Comment:
This check fails for the shifted Bessel. I think that the code for matching Bessel's equation should not be in this if block.

Reply:
I think it would be best to leave the power series code alone (revert the changes) unless there is a good reason to change it.

Reply:
> This check fails for the shifted Bessel. I think that the code for matching Bessel's equation should not be in this if block.

I will remove the Bessel matching from here but I think we can keep the changes in the series solution.

Reply:
Is any behaviour changed or is it just a cleanup of the code?